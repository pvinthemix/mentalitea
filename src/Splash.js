import React, { Component } from 'react';
import './styles/Main.scss';
import { ReactComponent as Cup } from './styles/images/cup.svg';
import { ReactComponent as SelectMood } from './styles/images/selectMood.svg';


export default class Splash extends Component {
  constructor() {
    super();
    this.state = {
      selectedMood: '',
      buttonText: 'Show All'
    };
  }

setUserSelectedMood = (event) => {
    this.setState({
      selectedMood: event.target.value,
      buttonText: 'Brew Tea'
    })
  }

returnTeaListResults = () => {
  this.props.getMoodFromDescriptor(this.state.selectedMood)
  this.props.toggleSplash()
}

render() {
  return (
    <div className="splash-page-container">
      <div className="instructions-container">
        <h2 className="instructions-heading">Instructions</h2>
          <p className="instructions-body"><span className="instruction-number">1)</span> Tell us how you're feeling at this moment.</p>
          <p className="instructions-body"><span className="instruction-number">2)</span> Click the "Brew Tea" button, and we will recommend teas accordingly.</p>
          <p className="instructions-body"><span className="instruction-number">3)</span> Feel free to click a tea to show more information about it.</p>
          <p className="instructions-body"><span className="instruction-number">4)</span> You can make adjustments or select a new mood at any time.</p>
      </div>
      <div className="select-mood-content">
        <SelectMood className="select-mood"/>
        <select className="user-selected-mood" onChange={this.setUserSelectedMood}>
          <option>Select your mood</option>
          {
            this.props.moods.reduce((allDescriptors, mood) => {
              return allDescriptors.concat(mood.descriptors);
            }, []).map((mood, index) => {
              return <option key={index} value={mood}>{mood}</option>;
            })
          }
          
        </select>
      </div>
      <div className="cup-container">
        <Cup className="cup"/>
        <button className="splash-page-button" onClick={this.returnTeaListResults}>
          {this.state.buttonText}
        </button>
      </div>
    </div>
  );
}
}