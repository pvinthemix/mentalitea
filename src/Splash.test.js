import React from 'react';
import Splash from './Splash';
import { shallow } from 'enzyme';

const moods = [{
  mood: "Stressed",
  descriptors: [
    "agitated",
    "on-edge",
    "worried",
    "tense"
  ],
  moodId: 1
}];

const getMoodFromDescriptorMock = jest.fn();
const toggleSplashMock = jest.fn();

describe('Splash', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Splash moods={moods}
        getMoodFromDescriptor={getMoodFromDescriptorMock}
        toggleSplash={toggleSplashMock}
      />
    );
  });

  it('should update the user\'s selected mood', () => {
    wrapper.find('.user-selected-mood').simulate('change', {target: {value: 'annoyed'}});
    expect(wrapper.state('selectedMood')).toEqual('annoyed');
  });

  it('should invoke toggleSplash and getMoodFromDescriptor when clicked', () => {
    wrapper.find('.splash-page-button').simulate('click');
    expect(getMoodFromDescriptorMock).toBeCalled();
    expect(toggleSplashMock).toBeCalled();
  });

  it('should display buttons, instructions, and svgs', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
});