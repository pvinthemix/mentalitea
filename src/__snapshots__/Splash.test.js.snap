// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`Splash should display buttons, instructions, and svgs 1`] = `
<div
  className="splash-page-container"
>
  <div
    className="instructions-container"
  >
    <h2
      className="instructions-heading"
    >
      Instructions
    </h2>
    <p
      className="instructions-body"
    >
      <span
        className="instruction-number"
      >
        1)
      </span>
       Tell us how you're feeling at this moment.
    </p>
    <p
      className="instructions-body"
    >
      <span
        className="instruction-number"
      >
        2)
      </span>
       Click the "Brew Tea" button, and we will recommend teas accordingly.
    </p>
    <p
      className="instructions-body"
    >
      <span
        className="instruction-number"
      >
        3)
      </span>
       Feel free to click a tea to show more information about it.
    </p>
    <p
      className="instructions-body"
    >
      <span
        className="instruction-number"
      >
        4)
      </span>
       You can make adjustments or select a new mood at any time.
    </p>
  </div>
  <div
    className="select-mood-content"
  >
    <ReactComponent
      className="select-mood"
    />
    <select
      className="user-selected-mood"
      onChange={[Function]}
    >
      <option>
        Select your mood
      </option>
      <option
        key="0"
        value="agitated"
      >
        agitated
      </option>
      <option
        key="1"
        value="on-edge"
      >
        on-edge
      </option>
      <option
        key="2"
        value="worried"
      >
        worried
      </option>
      <option
        key="3"
        value="tense"
      >
        tense
      </option>
    </select>
  </div>
  <div
    className="cup-container"
  >
    <ReactComponent
      className="cup"
    />
    <button
      className="splash-page-button"
      onClick={[Function]}
    >
      Show All
    </button>
  </div>
</div>
`;
